src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+
+    private Object[] elements;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfFull();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (checkAddIndex(index)) {
+            growIfFull();
+            elements = addResize(value, index);
+            size++;
+        }
+    }
 
+    private Object[] addResize(T value, int index) {
+        Object[] update = new Object[elements.length];
+        System.arraycopy(elements, 0, update, 0, index);
+        update[index] = value;
+        System.arraycopy(elements, index, update, index + 1,
+                size - index);
+        return update;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndex(index);
+        elements[index] = value;
 
     }
 
     @Override
-    public T remove(int index) {
-        return null;
+    public T remove(T element) {
+        for (int i = 0; i < size; i++) {
+            if (element == null && elements[i] == null
+                    || element != null && element.equals(elements[i])) {
```

Comment:
We can simplify this
```suggestion
            if (element == elements[I] || element != null && element.equals(elements[i])) {
```